cocotb/share/lib/gpi/GpiCommon.cpp

```diff
@@ -618,22 +605,24 @@ gpi_sim_hdl gpi_register_readwrite_callback(int (*gpi_function)(const void *),
 
 gpi_sim_hdl gpi_create_clock(gpi_sim_hdl clk_signal, const int period)
 {
-    GpiObjHdl *clk_hdl = sim_to_hdl<GpiObjHdl*>(clk_signal);
+    GpiObjHdl *clk_hdl = clk_signal;
     GpiClockHdl *clock = new GpiClockHdl(clk_hdl);
     clock->start_clock(period);
     return (gpi_sim_hdl)clock;
 }
 
 void gpi_stop_clock(gpi_sim_hdl clk_object)
 {
-    GpiClockHdl *clock = sim_to_hdl<GpiClockHdl*>(clk_object);
+    // TODO: Why are we pretending this is one type when it is another?
+    GpiClockHdl *clock = reinterpret_cast<GpiClockHdl*>(clk_object);
     clock->stop_clock();
     delete(clock);
 }
 
 void gpi_deregister_callback(gpi_sim_hdl hdl)
 {
-    GpiCbHdl *cb_hdl = sim_to_hdl<GpiCbHdl*>(hdl);
+    // TODO: Why are we preteding this is one type when it is another?
+    GpiCbHdl *cb_hdl = reinterpret_cast<GpiCbHdl*>(hdl);
     cb_hdl->m_impl->deregister_callback(cb_hdl);
 }
```

Comment:
Probably best to just introduce a `gpi_cb_hdl` and `gpi_clock_hdl` C typedef for this in a follow-up.

Reply:
Actually, I'll update this PR to include that change once #1429 is in